### DISCUSSION TITLE
CefSharp OffScreen has wrong values for width and height in window.screen where as CefSharp WinForms has correct values

### DISCUSSION THREAD

**Discussion by Girivasan** [cefsharp/CefSharp#4769] [Q&A]
Our app is a background application which records screen using navigator.mediaDevices apis. Since it is a background application, we opted for CefSharp OffScreen. On investigation, I found out that recorded screen resolution is reduced one and with lower quality.

My Windows laptop resolution is 1920 x 1080
![Screenshot 2024-03-20 233325](https://github.com/cefsharp/CefSharp/assets/8578815/089c565c-d3a2-4281-8ebd-6e0cf5323777)

I compared below values in CefSharp Minimal Example project (OffScreen vs WinForms)

OffScreen: Here resolution is 1366 x 768
![Screenshot 2024-03-26 155735](https://github.com/cefsharp/CefSharp/assets/8578815/62c8d52f-6ffb-48ff-996f-b4520aa766ae)

WinForms: Here resolution is 1920 x 1080 (resolution width = 1536 * 1.25 = 1920, resolution height = 864 * 1.25 = 1080) . These values align with chrome and firefox values.
![Screenshot 2024-03-26 160031](https://github.com/cefsharp/CefSharp/assets/8578815/fa311e93-50f0-41ce-850f-ae7cd65dcc1f)

What needs to be done to get correct screen resolution in CefSharp OffScreen similar to CefSharp Winforms ? Anyone please help

Or is there a way to run CefSharp Winforms as a background application without showing any UI ?

Additional note :

- I cannot preconfigure browser size because the same application needs to record multiple monitors
- In case of multiple monitors, resolution is reduced in the additional monitor in case of CefSharp Offscreen

**Comment by amaitland** [comment]
The default size is 1366 x 768

You can specify the size via the Size property 

https://github.com/cefsharp/CefSharp/blob/master/CefSharp.OffScreen/ChromiumWebBrowser.cs#L416

You can specify the scale via DeviceScaleFactor property

https://github.com/cefsharp/CefSharp/blob/master/CefSharp.OffScreen/ChromiumWebBrowser.cs#L436

It's up to you to configure the values to suit your requirements.
